controllers/operator/nmstate_controller.go

```diff
@@ -237,6 +239,34 @@ func (r *NMStateReconciler) applyHandler(instance *nmstatev1.NMState) error {
 	return r.renderAndApply(instance, data, "handler", true)
 }
 
+// webhookReplicaCount returns the number of replicas for the nmstate webhook
+// deployment based on the underlying infrastructure toplogy. It returns 2
+// values (and error):
+// 1. min. number of replicas
+// 2. number of desired replicas
+// 3. error
+func (r *NMStateReconciler) webhookReplicaCount(nodeSelector map[string]string) (int, int, error) { //nolint:gocritic
+	const (
+		multiNodeClusterReplicaCountDesired = 2
+		multiNodeClusterReplicaMinCount     = 1
+
+		singleNodeClusterReplicaCountDesired = 1
+		singleNodeClusterReplicaMinCount     = 0
```

Comment:
Without webhook installed NNCP cannot be applied since MutatingWebhookConfiguration is deployed.